### PR TITLE
Rework Security documents for Kyma 2.0

### LINKED ISSUE #10654: Update the document about OAuth2 server profiles
**Description**

The [OAuth2 server profiles](https://kyma-project.io/docs/master/components/security#configuration-o-auth2-server-profiles) document in its current form contains outdated and misleading information. It does not contain info about OAuth2 server working in the [production profile](https://kyma-project.io/docs/master/root/kyma#installation-overview-profiles) per se, but rather info on different modes of OAuth2 server configuration recommended for production purposes. Values for production profile are also included but are mixed with other production-recommended values for OAuth2 server configuration, that's why it's quite misleading. Values specific for the production profile per se should be removed from this document as a follow-up to [this](https://github.com/kyma-project/kyma/issues/9902) issue for describing profiles in a generic way, and only the configuration part should be left in this doc. 

Since it's difficult to differentiate which values are specific for the production profile and which are a part of the OAuth2 server configuration, some help from the developer's side will be needed to update this document.

**Area**

area/documentation
area/security

**Reasons**

The document in its current form is outdated and provides misleading info about production profile / OAuth2 server configuration.

**Assignees**

@kyma-project/technical-writers

**Related issues**

https://github.com/kyma-project/kyma/issues/9238
https://github.com/kyma-project/kyma/issues/9902

### PR DESCRIPTION
**Pull request by mmitoraj**
**Description**

Changes proposed in this pull request:

- Update diagrams
- Remove outdated details
- Update all docs in accordance with Kyma 2.0 scope

**Related issue(s)** 
Resolves #10654, #11723, #10149

### CODE REVIEW

**Comment by mmitoraj** [comment]
This is the list of documents to look at in the preview:

1. Tutorials -> Update TLS certificate
2. Operation guides -> Security -> Overview
3. Operation guides -> Security -> Authentication in Kyma
4. Operation guides -> Security -> Authorization in Kyma
5. Operation guides -> Security -> Access Kyma
6. Operation guides -> Security -> Ingress and Egress traffic
7. Operation guides -> Security -> OAuth2 server customization and operations
8. Operation guides -> Security -> Useful links
9. Operation guides -> Troubleshooting -> Issues with certificates on Gardener
10. Technical reference -> Configuration parameters -> Cluster users chart
11. Technical reference -> Configuration parameters -> ORY chart (not Security but API Exposure - WIP)

File: docs/04-operation-guides/security/07-sec-useful links.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+title: Useful links
+---
+
+If you're interested in learning more about Security in Kyma, follow these links to:
+
+- Perform some simple and more advanced tasks:
+  - [Update TLS certificate](../../03-tutorials/sec-01-tls-certificates-security.md).
+
+- Troubleshoot Security-related issues when:
+  - [Issues with certificates on Gardener](../troubleshooting/sec-02-certificates-gardener.md) come up.
```
Comment (werdes72) [inline docs/04-operation-guides/security/07-sec-useful links.md:11]: This link is not working.

Comment (werdes72) [reply docs/04-operation-guides/security/07-sec-useful links.md:11]: Seems like it should be:
`troubleshooting/sec-01-certificates-gardener`

**Review by werdes72 — CHANGES_REQUESTED** [review]

**Review by werdes72 — APPROVED** [review]

File: docs/04-operation-guides/security/02-sec-authentication-in-kyma.md [hunk 2]
```diff
@@ -37,8 +37,8 @@ The diagram shows the user authentication flow:
 
 ## Service-to-service authentication
 
-As Kyma is build on top of Istio Service Mesh, service-to-service authentication and encryption is enabled with [Istio MutualTLS](https://istio.io/latest/docs/concepts/security/#mutual-tls-authentication). For details, read the [Kyma-specific Istio configuration](/components/service-mesh/#details-istio-setup-in-kyma-kyma-specific-configuration) documentation.
+As Kyma is build on top of Istio Service Mesh, service-to-service authentication and encryption is enabled with [Istio MutualTLS](https://istio.io/latest/docs/concepts/security/#mutual-tls-authentication). For details, read the [Kyma-specific Istio configuration](../../01-overview/02-main-areas/service-mesh/smsh-02-default-istio-setup-in-kyma.md#kyma-specific-configuration) documentation.
```
Comment (NHingerl) [inline docs/04-operation-guides/security/02-sec-authentication-in-kyma.md:40]: ... is built...

File: docs/04-operation-guides/security/02-sec-authentication-in-kyma.md [hunk 3]
```diff
@@ -37,8 +37,8 @@ The diagram shows the user authentication flow:
...
 
 ## User-to-service authentication
 
-Kyma uses a custom [API Gateway](/components/api-gateway/#overview-overview) component that is build on top of [ORY Oathkeeper](https://www.ory.sh/oathkeeper/docs/). The API Gateway allows exposing user applications within the Kyma environment and secures them if necessary. You can then access the secured resources using [authentication options](/components/api-gateway/#architecture-architecture-request-flow).
+Kyma uses a custom [API Gateway](../../01-overview/02-main-areas/api-exposure/apix-01-api-gateway.md) component that is build on top of [ORY Oathkeeper](https://www.ory.sh/oathkeeper/docs/). The API Gateway allows exposing user applications within the Kyma environment and secures them if necessary. You can then access the secured resources using [authentication options](../../05-technical-reference/03-architecture/apix-01-architecture-apigateway.md#request-flow).
```
Comment (NHingerl) [inline docs/04-operation-guides/security/02-sec-authentication-in-kyma.md:44]: ... is built ...

File: docs/03-tutorials/sec-01-tls-certificates-security.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+    3. Trigger the update process. Run:
+
+      ```bash
+      kyma deploy -s $KYMA_VERSION
+      ```
+
+      The process is complete when you see the `Kyma installed` message.
+
+    4. Add the newly generated certificate to the trusted certificates of your OS. For MacOS, run:
```
[the remark this answers is not in the document]
Comment (mmitoraj) [reply docs/03-tutorials/sec-01-tls-certificates-security.md:68]: :(

**Pull request merged**